tools/wake-format/emitter.cpp

```diff
@@ -65,6 +65,21 @@ static inline bool is_expression(cst_id_t type) {
          type == CST_INTERPOLATE || type == CST_MATCH;
 }
 
+// a floating comment is a comment bound to another comment
```

Comment:
Ah, yeah. I don't think it's possible to remove the comment from the tree without a significant change to the underlying data structure (CSTElement). It would certainly be a nicer design to just remove the bound comments if we could. Though it would only be an improvement assuming floating comments are top level only. As long as comments can appear anywhere in the tree we will have to handle them specifically. 

That said, binding a "floating" comment to itself is fundamentally a hack but its a shorter term hack that can be unwound fairly easily if a better solution manifests itself in the future